src/System.Private.CoreLib/shared/System/Memory.cs

```diff
@@ -407,18 +454,7 @@ public bool Equals(Memory<T> other)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override int GetHashCode()
         {
-            return _object != null ? CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode()) : 0;
+            return (_object != null) ? ReadOnlyMemory<T>.CombineHashCodes(RuntimeHelpers.GetHashCode(_object), _index, _length) : 0;
```

Comment:
Why not just use `HashCode.Combine` ?

Reply:
`HashCode.Combine` is randomized because it pessimistically assumes its inputs are attacker-controlled. The `Memory<T>` object doesn't really qualify as attacker-controlled, so it can use a more performance-oriented routine.

Reply:
That's a good point, especially since the default implementation is referential equality. We can make them use the slower `HashCode` helpers for now for simplicity, and it leaves the door open for us to make targeted perf improvements to these methods in the future if the need arises.